app/src/main/java/com/techcourse/service/AppUserService.java

```diff
@@ -0,0 +1,39 @@
+package com.techcourse.service;
+
+import com.techcourse.dao.UserDao;
+import com.techcourse.dao.UserHistoryDao;
+import com.techcourse.domain.User;
+import com.techcourse.domain.UserHistory;
+
+import java.util.NoSuchElementException;
+
+public class AppUserService implements UserService {
+
+    private final UserDao userDao;
+    private final UserHistoryDao userHistoryDao;
+
+    public AppUserService(final UserDao userDao, final UserHistoryDao userHistoryDao) {
+        this.userDao = userDao;
+        this.userHistoryDao = userHistoryDao;
+    }
+
+    @Override
+    public User findById(final long id) {
+        return userDao.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("id 값으로 해당하는 User 를 찾을 수 없습니다."));
```

Comment:
```suggestion
                .orElseThrow(() -> new NoSuchElementException(String.format("id 값으로 해당하는 User 를 찾을 수 없습니다. 입력값: %s", id)));
```

1. 클라이언트가 어떤 입력값을 보냈을 때 찾을 수 없었는지 추적할 수 있도록 메세지에 입력값을 남기세요.
2. NoSuchElementException 가 적절한가요? Element라는 용어는 자바 컬렉션에서 사용되는걸로 보이는데, 적절한 예외일까요?

Reply:
1. 넵 디버깅하기가 쉬워지겠네요.
2. 오.. 그 점은 고려하지 못했습니다. 앞으로 예외의 패키지도 의식적으로 보는 연습이 필요할것 같습니다